### ISSUE TITLE
[BUG] DefaultAzureCredentials - Azure Identity fails (local) with v 1.10.x but works with 1.9.0

### ISSUE THREAD

**Issue by gkrishna07** [Azure/azure-sdk-for-net#39532]
### Library name and version

Azure.Identity v1.10.3

### Describe the bug

This code works in Azure Identity 1.9.0 but fails in v 1.10.x fails: 

```
var credentials = new DefaultAzureCredential();
var configKeys = settings["AppSettings:ConfigKeys"]?.Split('|');
config.AddAzureAppConfiguration(options =>
{
    // 1. Connect to App Configuration using the default credentials
    options.Connect(new Uri(settings.GetValue<string>("AppSettings:AppConfig")), credentials);
    // 2. Set security for key vault as there are secrets to be resolved
    options.ConfigureKeyVault(kv =>
    {
        kv.SetCredential(credentials);
    });
```

Upgraded Azure Identity from version 1.9.0 to 1.10.0 (also tried upto recent 1.10.3), facing the below error from local environment.
![Screenshot 2023-10-26 235311](https://github.com/Azure/azure-sdk-for-net/assets/149074998/ddd5a1b2-a952-4054-9d79-d5e46cee20e7)

### Expected behavior

Authentication works without exception.

### Actual behavior

When upgraded to any 1.10.x version facing the below error in local environment.

```
Azure.Identity.AuthenticationFailedException: 'ManagedIdentityCredential authentication failed: Service request failed.
Status: 404 (Not Found)

Content:
<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
		 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
  <title>404 - Not Found</title>
 </head>
 <body>
  <h1>404 - Not Found</h1>
 </body>
</html>


Headers:
Date: Thu, 26 Oct 2023 18:02:55 GMT
Server: EC2ws
Connection: close
Content-Type: text/html
Content-Length: 339

See the troubleshooting guide for more information. https://aka.ms/azsdk/net/identity/managedidentitycredential/troubleshoot'

```

### Reproduction Steps

To reproduce the issue upgrade to Azure.Identity latest version

### Environment

 <TargetFramework>net6.0</TargetFramework>
<DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>

<PackageReference Include="AutoMapper" Version="10.1.1" />
<PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="8.1.1" />
<PackageReference Include="Azure.Identity" Version="1.10.0" />
<PackageReference Include="Azure.Security.KeyVault.Certificates" Version="4.5.0" />
<PackageReference Include="Azure.Security.KeyVault.Keys" Version="4.5.0" />
<PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.5.0" />

AKS Hosted on a docker-compose linux container - Works with managed identity as expected, but fails with DefaultAzureCredentials in local.

 Microsoft Visual Studio Professional 2022 (64-bit) 
Version 17.7.6 
Verified local token cache in Sync from VS 2022 IDE Account Settings or Tools->Options->Azure Service Authentication

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by christothes** [comment]
Hi @gkrishna07 - Would you mind reproducing this with [logging enabled](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md#logging) on the credential and sharing the logs? It might also be interesting to compare the logs between the working and non-working example.

**Comment by gkrishna07** [comment]
Authentication Failed. ManagedIdentityCredential authentication failed: Service request failed.
Status: 404 (Not Found)

Content:
<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
                 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
  <title>404 - Not Found</title>
 </head>
 <body>
  <h1>404 - Not Found</h1>
 </body>
</html>


Headers:
Date: Fri, 27 Oct 2023 09:05:36 GMT
Server: EC2ws
Connection: close
Content-Type: text/html
Content-Length: 339

See the troubleshooting guide for more information. https://aka.ms/azsdk/net/identity/managedidentitycredential/troubleshoot

**Comment by gkrishna07** [comment]
@christothes  attached mentioned logs to previous comment - fyr,

With DefaultAzureCredentials acquiring token it should fall back to next credential 
if ManagedIdentityCredential.GetToken failed it should next move on to attempt VisualStudioCredential.GetToken which is not occurring with any 1.10.x versions

**Comment by christothes** [comment]
Hi @gkrishna07 
I believe this behavior is due to a fix made since version 1.9.0 was released. [Status code 404 is a retriable error code](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance) for the IMDS endpoint. This means that we consider this to be a valid response and we fail rather than continue on to other credentials. 

Typically, in a development environment the IMDS endpoint should not respond at all but sometimes there are services or proxies on the network that reply to these requests. 

The recommended workaround is either to prevent the service in your environment from responding to requests to `http://169.254.169.254/metadata/identity/oauth2/token` or to conditionally set [ExcludeManagedIdentityCredential](https://github.com/Azure/azure-sdk-for-net/blob/70cbe220f44c7d430baa9535dac0ec59d0dd9504/sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs#L224) to `true` in the DefaultAzureCredentialOptions.

**Comment by toni-ar** [comment]
> Hi @gkrishna07 I believe this behavior is due to a fix made since version 1.9.0 was released. [Status code 404 is a retriable error code](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance) for the IMDS endpoint. This means that we consider this to be a valid response and we fail rather than continue on to other credentials.
> 
> Typically, in a development environment the IMDS endpoint should not respond at all but sometimes there are services or proxies on the network that reply to these requests.
> 
> The recommended workaround is either to prevent the service in your environment from responding to requests to `http://169.254.169.254/metadata/identity/oauth2/token` or to conditionally set [ExcludeManagedIdentityCredential](https://github.com/Azure/azure-sdk-for-net/blob/70cbe220f44c7d430baa9535dac0ec59d0dd9504/sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs#L224) to `true` in the DefaultAzureCredentialOptions.

Had the same issue, can confirm that `ExcludeManagedIdentityCredential` workaround works.
